File: LENGUAJES/PYTHON/Ejercicio10.py
#LISTAS

def mostrar(materias):
    if materias:
        print(f"Yo estudio {len(materias)} materias.")
        print("Las materias son:")
        for materia in materias:
            print(f"- {materia}")
    else:
        print("No hay materias para mostrar.")

def agregar_materia(materias):
    nueva_materia = input("Ingresa el nombre de la nueva materia: ").strip()
    if nueva_materia:
        materias.append(nueva_materia)
        print(f"La materia '{nueva_materia}' ha sido agregada.")
    else:
        print("El nombre de la materia no puede estar vacío.")

def main():
    lista = ["Matematicas", "Fisica", "Quimica", "Historia", "Lengua"]
    mostrar(lista)

    while True:
        continuar = input("¿Deseas agregar una nueva materia? (s/n): ").strip().lower()
        if continuar == 's':
            agregar_materia(lista)
            mostrar(lista)
        elif continuar == 'n':
            print("Gracias por usar el programa. ¡Hasta luego!")
            break
        else:
            print("Opción no válida. Por favor, ingresa 's' o 'n'.")
main()

File: LENGUAJES/GO/Ejercicio10.go
//LISTAS
package main

import (
	"fmt"
	"strings"
)

func main() {
	lista := []string{"Matematicas", "Fisica", "Quimica", "Historia", "Lengua"}
	mostrar(lista)

	for {
		var continuar string
		fmt.Print("¿Deseas agregar una nueva materia? (s/n): ")
		fmt.Scanln(&continuar)
		continuar = strings.ToLower(continuar)

		if continuar == "s" {
			agregarMateria(&lista)
			mostrar(lista)
		} else if continuar == "n" {
			fmt.Println("Gracias por usar el programa. ¡Hasta luego!")
			break
		} else {
			fmt.Println("Opción no válida. Por favor, ingresa 's' o 'n'.")
		}
	}
}

func mostrar(materias []string) {
	if len(materias) > 0 {
		fmt.Printf("Yo estudio %d materias.\n", len(materias))
		fmt.Println("Las materias son:")
		for _, materia := range materias {
			fmt.Printf("- %s\n", materia)
		}
	} else {
		fmt.Println("No hay materias para mostrar.")
	}
}

func agregarMateria(materias *[]string) {
	var nuevaMateria string
	fmt.Print("Ingresa el nombre de la nueva materia: ")
	fmt.Scanln(&nuevaMateria)
	nuevaMateria = strings.TrimSpace(nuevaMateria)

	if nuevaMateria != "" {
		*materias = append(*materias, nuevaMateria)
		fmt.Printf("La materia '%s' ha sido agregada.\n", nuevaMateria)
	} else {
		fmt.Println("El nombre de la materia no puede estar vacío.")
	}
}